game-core/src/main/java/games/strategy/engine/data/MoveDescription.java

```diff
@@ -26,6 +26,9 @@ public MoveDescription(
       final Map<Unit, Collection<Unit>> dependentUnits) {
     super(Collections.unmodifiableCollection(units));
     this.route = Preconditions.checkNotNull(route);
+    if (route.getEnd() == null || route.getStart() == null || route.hasNoSteps()) {
```

Comment:
A single territory route is not valid for a move description? Can you clarify if that is fully intended.

``` route.getStart() == null```  is guaranteed by route?

Side-note, kinda nice to have the single constructor, but it looks like the start is asserted as not null:
```
  public Route(final Territory start, final List<Territory> territories) {
    this.start = checkNotNull(start);
    for (final Territory t : territories) {
      add(t);
    }
  }

  public Route(final List<Territory> territories) {
    this(territories.get(0), territories.subList(1, territories.size()));
  }

  public Route(final Territory start, final Territory... territories) {
    this(start, List.of(territories));
  }
```

It also seems like `getEnd()` can not be null either (start is guaranteed to be non-null, if steps is non-empty then we guarantee that we do not add any null elements to it):

```
  public Territory getEnd() {
    if (steps.isEmpty()) {
      return start;
    }
    return steps.get(steps.size() - 1);
  }
```
```
  /** Add the given territory to the end of the route. */
  private void add(final Territory territory) {
    checkNotNull(territory);
    if (territory.equals(start) || steps.contains(territory)) {
      throw new IllegalArgumentException(
          String.format(
              "Loops not allowed in steps, route: %s, new territory: %s", this, territory));
    }
    steps.add(territory);
  }
```

Reply:
Thanks for checking, agreed. Removing these.

Keeping route.hasNoSteps() check, since I think that's still valid. My understanding is each move must have steps (i.e. a unit is moving from somewhere to somewhere). Let me know if you think that's not the case.

Reply:
> My understanding is each move must have steps (i.e. a unit is moving from somewhere to somewhere). Let me know if you think that's not the case.

Sounds reasonable, but TBH I'm not sure. Would you mind double checking on this?

Reply:
I wasn't able a way to create such a move from the UI. For example, in MovePanel.updateUnitsThatCanMoveOnRoute(), it clears the move if the route has no steps.

In terms of whether a move is "valid", well MoveValidator does consider it as valid per:

```
    final MoveValidationResult result = new MoveValidationResult();
    if (route.hasNoSteps()) {
      return result;
    }
```

But I think nothing would happen in this case when the move is performed. And I also played a full AI game with no such routes being generated. So I think this is safe.